src/app/components/publiclists/challengelist/challengelist.component.ts

```diff
@@ -119,15 +132,38 @@ export class ChallengelistComponent implements OnInit {
     if (this.authService.isLoggedIn()) {
       this.isLoggedIn = true;
     }
+
+    this.authServiceSubscription = this.authService.change.subscribe((authState) => {
+      this.isLoggedIn = authState.isLoggedIn;
```

Comment:
Define `'/auth/login'`, `/challenges/me'` as constants above. Check other such instances in this file/PR.

Reply:
done